taichi/ir/ir.h

```diff
@@ -872,6 +872,18 @@ class Block : public IRNode {
   TI_DEFINE_ACCEPT
 };
 
+class DelayedIRModifier {
+ private:
+  std::vector<std::pair<Stmt *, VecStatement>> to_insert_before;
+  std::vector<Stmt *> to_erase;
+
+ public:
+  void erase(Stmt *stmt);
+  void insert_before(Stmt *old_statement, std::unique_ptr<Stmt> new_statement);
+  void insert_before(Stmt *old_statement, VecStatement &&new_statements);
+  bool modify_ir();
```

Comment:
Should we check if `to_insert_before` and `to_erase` to be more fool-proof? Just in case someone forgets to call `modify_ir()`.

Reply:
Sure! I'm not sure if it's safe to write `TI_ASSERT` in destructors though.

Reply:
Should be fine. Virtual functions are what should be avoided in dtors.